philips-hue-adapter.js

```diff
@@ -577,17 +605,24 @@ class PhilipsHueDevice extends Device {
 
     if (this.properties.has('buttonOn')) {
       const buttonEvent = device.state.buttonevent;
+      const lastUpdated = device.state.lastupdated;
+      const lastUpdatedProp = this.properties.get('lastUpdated');
+      const isPressed = lastUpdatedProp.value !== lastUpdated;
+
       for (const buttonType in HUE_DIMMER_SWITCH_BUTTONS) {
         if (HUE_DIMMER_SWITCH_BUTTONS.hasOwnProperty(buttonType)) {
           const buttonInfo = HUE_DIMMER_SWITCH_BUTTONS[buttonType];
           const buttonProp = this.properties.get(buttonType);
-          const pressed = buttonEvent >= buttonInfo.start &&
-            buttonEvent < buttonInfo.off;
-          if (buttonProp.value !== pressed) {
-            buttonProp.setCachedValueAndNotify(pressed);
+          const isButton = buttonEvent >= buttonInfo.initialPress &&
+            buttonEvent <= buttonInfo.longRelease;
+
+
+          if (buttonProp.value !== (isPressed && isButton)) {
```

Comment:
Why are you setting the value of the property to `isPressed && isButton`? Shouldn't it just be `isButton`?

Reply:
**lastUpdated** tells you WHEN a button is pressed whereas **buttonEvent** tells you WHICH button was pressed. So by monitoring changes to **lastUpdated** we are notified of individual button presses. However **buttonProp** is set to true only when the button isPressed (**lastUpdated** has changed) AND **isButton**(**buttonEvent** falls between the range of the buttonInfo being scanned). Hope that makes sense.